refactor: MyPage(마이페이지) Next.js + TS 마이그레이션

whozbebop opened the pull request:
## 변경 사항

- 마이페이지 라우팅을 App Router 기반으로 전환 (`app/mypage/*`)
- 기존 React Router 요소(`to`, `NavLink`) → Next.js `href`, `Link`로 전환
- MyPage 관련 페이지(Profile, Enrolled, InstructorCourses) TSX 마이그레이션
- Enrolled/Instructor 타입 정의 분리 및 서비스 함수 반환 타입 명확화

## 리뷰 필요

- App Router 적용 방식 및 기존 구조 유지 방향성 문제 없는지 확인
- 타입 정의(Enrolled / Instructor) 구조 적절성 검토
- MyPage 내부 경로 구조(/mypage/profile, /mypage/enrolled, /mypage/instructor) 확인
- Firebase 연동 전 임시 로그인(mockUser) 처리 방식 적정성 확인

close #43

whozbebop committed:
update migration

whozbebop committed:
update migration

whozbebop committed:
update migration

whozbebop committed:
feat: auth.ts, enrolled.ts, user.ts 추가

whozbebop committed:
feat: useAuthState.ts / authService.ts, enrolledService.ts, userService.ts 수정

whozbebop committed:
feat: MyPage 관련 마이그레이션

whozbebop committed:
Merge branch 'refactor/#43' of https://github.com/20250915-POTENUP-FE-1/LXP_lern-p into refactor/#43

whozbebop committed:
Merge branch 'migration/next' of https://github.com/20250915-POTENUP-FE-1/LXP_lern-p into refactor/#43

whozbebop committed:
feat: instructor.ts 강사가 개설한 강좌 타입 정의

whozbebop committed:
feat: MyPage 관련 마이그레이션
 - MyPage 레이아웃 AppShell 추가
 - 내 정보, 수강 중인 강좌, 장바구니 수정
 - 내가 등록한 강좌 수정
 - 사이드바 active 클래스 추가
 - 가상 로그인 관련 삭제

whozbebop committed:
chore: MyPage 관련 파일 정리
 - console.log 수정
 - js 파일 삭제
 - 주석 삭제

whozbebop committed:
Merge branch 'migration/next' of https://github.com/20250915-POTENUP-FE-1/LXP_lern-p into refactor/#43

kimsuyeon0916 commented on src/domains/user/types/auth.ts:
migration/next pull 받으시고 제거 부탁드려요!

kimsuyeon0916 commented on src/app/(user)/mypage/cart/page.tsx:
```diff
@@ -1,3 +1,16 @@
+import Link from 'next/link';
+import styles from '@/domains/user/pages/MyPageSections.module.css';
+
+export default function Cart() {
```
페이지에 관련된 것들은 접미사 Page 붙이는게 컨벤션~~

kimsuyeon0916 commented on src/domains/user/types/enrolled.ts:
Course 관련 타입들을 이용해서 정리 부탁드려요! by 유틸리티 타입

kimsuyeon0916 commented on src/domains/user/types/instructor.ts:
Course 관련 타입들을 이용해서 정리 부탁드려요! by 유틸리티 타입

kimsuyeon0916 commented on src/app/(user)/mypage/enrolled/page.tsx:
```diff
@@ -1,3 +1,90 @@
+import { getEnrolledCourses } from "@/domains/user/services/enrolledService";
+import type { EnrolledCourse } from "@/domains/user/types/enrolled";
+
+export default function Enrolled() {
```
```suggestion
export default function EnrolledCoursesPage() {
```

whozbebop committed:
feat: shared hooks/utils 타입 적용
- useModal.js → useModal.ts 전환 및 명시적 타입 추가
- validateForm.js → validateForm.ts 전환, FormValue 타입 정의
- 폼 검증 및 모달 훅 타입 안정성 강화

whozbebop committed:
refactor: 분리 정돈 Server/Client components
- 불필요한 "use client" 지시어 제거
- 상태/인터랙션 있는 컴포넌트만 Client Component로 유지
- 레이아웃 및 정적 UI 컴포넌트는 Server Component로 전환
- Next.js App Router 가이드라인 준수

whozbebop committed:
chore: remove unused pages & CSS 모듈 정리
- MyPage.module.css 등 pages 기반 스타일 삭제
- Instructor/ClientPage 등 사용되지 않는 파일 제거
- 프로젝트 전체 불필요 리소스 정돈

whozbebop committed:
chore: CSS 모듈 정리

whozbebop committed:
refactor: formatUserDate 유틸 Firestore 의존성 제거 및 순수 Date 처리로 단순화

- FirestoreTimestamp 타입 제거
- timestamp 입력 타입을 string | Date 로 축소
- DB 종속성 제거 및 유틸 재사용성 향상

whozbebop committed:
refactor: 클라이언트 컴포넌트 도메인 계층으로 이동 및 MyPage CSS 재배치

- Client 전용 페이지 파일을 src/domains/user/pages 로 이동
- App Router 기반 라우트 구조에 맞게 MyPage CSS 모듈 정리
- 라우팅/스타일 파일 구조 일관성 개선

whozbebop committed:
chore: 불필요한 wrapper(section) 제거
